Codacy and CircleCI reviews of the tools

ehavazli opened the pull request:
I made a general review of the tools with respect to codacy suggestions. There are still some issues in some parts but majority of the files are rated A. Also CircleCI cuild is working now.

- CircleCI build is passed and includes the badge at the top of the README.md
- Codacy project rating badge is on top of README.md.

Suggested future work:
- get rid of subprocess completely.
- get rid of duplications (making a general definition of arguments, which are the most duplicated part, in a separate file can be an option)

ehavazli committed:
Merge pull request #1 from aria-tools/dev

Fixed minor bug in argparse for num_thread default. (#76)

ehavazli committed:
Merge pull request #2 from aria-tools/dev

Bug fixes in prep_dem and prep_mask functions (#79)

ehavazli committed:
New tool:
- create bounding box files to use with ARIA-tools from Google Earth polygon kml files

ehavazli committed:
Merge branch 'dev' of https://github.com/ehavazli/ARIA-tools into dev

ehavazli committed:
fixed typo

ehavazli committed:
fixed typo

ehavazli committed:
Merge pull request #3 from aria-tools/dev

update upstream

ehavazli committed:
Merge pull request #4 from aria-tools/dev

update

ehavazli committed:
quality review:
- Changes with respec to Codacy review

ehavazli committed:
quality review:
-Changes with respect to Codacy review

ehavazli committed:
quality review:
- Rename files with respect to Codacy review

ehavazli committed:
Codacy review

ehavazli committed:
Linux_source_build.md

ehavazli committed:
Delete Linux_source_build.md

ehavazli committed:
quality review:
- implement some Codacy reviews

ehavazli committed:
Merge branch 'dev' of https://github.com/ehavazli/ARIA-tools into dev

ehavazli committed:
Merge pull request #6 from aria-tools/dev

rebase/update fork

ehavazli committed:
Codacy quality review

ehavazli committed:
Merge branch 'dev' of https://github.com/ehavazli/ARIA-tools into dev

ehavazli committed:
fixed typo

ehavazli committed:
Codacy review
 - removed unused module

ehavazli committed:
Codacy review suggestions

ehavazli committed:
Codacy review

ehavazli committed:
Codacy review

ehavazli committed:
Codacy review suggestions

codacy-badger committed:
Add Codacy badge

ehavazli committed:
Merge pull request #7 from codacy-badger/codacy-badge

Add a Codacy badge to README.md

ehavazli committed:
Update README.md

ehavazli committed:
Update README.md

ehavazli committed:
Shell True in subprocess.call

ehavazli committed:
Merge branch 'dev' of https://github.com/ehavazli/ARIA-tools into dev

dbekaert commented on README.md:
```diff
@@ -2,6 +2,8 @@
 
 [![Language](https://img.shields.io/badge/python-3.5%2B-blue.svg)](https://www.python.org/)
 [![License](https://img.shields.io/badge/license-GPL-yellow.svg)](https://github.com/aria-tools/ARIA-tools/blob/master/LICENSE)
+[![Codacy Badge](https://api.codacy.com/project/badge/Grade/b69726c41ff9498882088afe57fe8391)](https://app.codacy.com/app/ehavazli/ARIA-tools?utm_source=github.com&utm_medium=referral&utm_content=ehavazli/ARIA-tools&utm_campaign=Badge_Grade_Settings)
```
This is pointing to your own app for codacy and the circleci

ehavazli replied:
I made the changes

dbekaert commented on tools/ARIAtools/ARIAProduct.py:
```diff
@@ -182,6 +180,7 @@ def __mappingData__(self, file, rmdkeys, sdskeys):
         datalyr_dict={}
 
         # Setup rdrmetadata_dict
+        # for i, j in enumerate(rdrmetakeys):
```
not sure this is needed here.

ehavazli replied:
removed i from the loop and the commented out line as well.

dbekaert commented:
it does not look like circle ci is being ran.

sssangha commented:
@ehavazli Are you sure that the .circleci file exists on your end? In the past, I think I encountered a similar circleci failure when I pushed changes from a v1 checkout (with no .circleci file) to dev

ehavazli commented:
> @ehavazli Are you sure that the .circleci file exists on your end? In the past, I think I encountered a similar circleci failure when I pushed changes from a v1 checkout (with no .circleci file) to dev

@sssangha, yes I do have .circleci folder with config.yml in it on my end. How did you solve this issue?

ehavazli committed:
Update README.md

ehavazli committed:
Update ARIAProduct.py

dbekaert commented on README.md:
```diff
@@ -2,8 +2,8 @@
 [![License](https://img.shields.io/badge/license-GPL-yellow.svg)](https://github.com/aria-tools/ARIA-tools/blob/master/LICENSE)
-[![Codacy Badge](https://api.codacy.com/project/badge/Grade/b69726c41ff9498882088afe57fe8391)](https://app.codacy.com/app/ehavazli/ARIA-tools?utm_source=github.com&utm_medium=referral&utm_content=ehavazli/ARIA-tools&utm_campaign=Badge_Grade_Settings)
-[![CircleCI](https://circleci.com/gh/ehavazli/ARIA-tools.svg?style=svg)](https://circleci.com/gh/ehavazli/ARIA-tools)
+[![Codacy Badge](https://api.codacy.com/project/badge/Grade/ab9ba02d44c440fdac034cb144278b33)](https://www.codacy.com/app/ehavazli/ARIA-tools_2?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=aria-tools/ARIA-tools&amp;utm_campaign=Badge_Grade)
```
The codacy badge still points to you. Is that correct?

ehavazli replied:
I think it is because I run it from my account. if you look at the last part of the line it says aria-tools/ARIA-tools

```
[![Codacy Badge](https://api.codacy.com/project/badge/Grade/ab9ba02d44c440fdac034cb144278b33)](https://www.codacy.com/app/ehavazli/ARIA-tools_2?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=aria-tools/ARIA-tools&amp;utm_campaign=Badge_Grade
```

dbekaert commented on tools/ARIAtools/ARIAProduct.py:
```diff
@@ -180,10 +180,9 @@ def __mappingData__(self, file, rmdkeys, sdskeys):
+        for j in enumerate(rdrmetakeys):
             try: #If layer expected
-                rdrmetadata_dict[radarkeys[rmdkeys.index(j)]]=rdrmetadata[j][0]
+                rdrmetadata_dict[radarkeys[rmdkeys.index(j[1])]]=rdrmetadata[j[1]][0]
```
@sssangha please confirm

sssangha replied:
@ehavazli ok, it looks like enumeration is no longer needed. So why not just have a classic "for" loop? In that case, there isn't a need to catch the 1st index for each iterated object?:
```
for i in rdrmetadatakeys: 
    try:
        rdrmetadata_dict[radarkeys[rmdkeys.index(j)]]=rdrmetadata[j][0]
```

I see this also a couple of other "for" loops in productPlot and extractProduct. Let me know if this makes sense, and I could go ahead and adjust it accordingly

ehavazli replied:
yes please go ahead, codacy was complaining about having two variables and looping around range(len()).

ehavazli committed:
change enumarate

ehavazli committed:
Merge branch 'dev' of https://github.com/ehavazli/ARIA-tools into dev

sssangha commented on tools/ARIAtools/extractProduct.py:
```diff
@@ -179,12 +178,12 @@ def prep_mask(product_dict, maskfilename, bbox_file, prods_TOTbbox, proj, amp_th
 
         ###Make lakes/ponds union VRT
-        os.system('ogrmerge.py -o ' + os.path.join(workdir,'watermsk_lakes.vrt') + ''.join(_world_watermask[1::2]) + ' -field_strategy Union -f VRT -single')
+        subprocess.call('ogrmerge.py -o ' + os.path.join(workdir,'watermsk_lakes.vrt') + ''.join(_world_watermask[1::2]) + ' -field_strategy Union -f VRT -single',shell=True)
```
@ehavazli is there a particular reason why we should system from an "os" call to "subprocess"?

ehavazli replied:
review packages always flag os.system() as an issue. while the same is correct for subprocess, it is raising less flags. As I mentioned in the main comment we should also move away from subprocess as well.

sssangha committed:
Adjusted enumeration in for loops, remove need for subprocess.

ehavazli committed:
switched to enumerate

range(len()) switched enumerate

ehavazli committed:
remove sys and pass

remove sys and pass

ehavazli committed:
remove PhaseUnwrap module

ehavazli committed:
Merge pull request #8 from sssangha/ehavazli-dev

Adjusted enumeration in for loops, remove need for subprocess.

dbekaert commented:
@sssangha does this all look good according to you?

Now that UNAVCO has been completed we can probably start merging items again and also make an increment to V1.

ehavazli commented:
> @sssangha does this all look good according to you?
> 
> Now that UNAVCO has been completed we can probably start merging items again and also make an increment to V1.

there are a couple more points I am working on right now and will commit in an hour or so

ehavazli committed:
Codacy review:

- ARIAProduct.py:     removed trailing whitespaces
- productPlot.py:     removed trailing whitespaces
- unwrapStitching.py: removed unused modules (Edge)
- ariaDownload.py:    removed subprocess usage
                      'ASFDataDload.py' file written out and imported within the code to avoid subprocess usage.

ehavazli committed:
removed unused module

unwrapStitching.py: Vertex
ariaDownload.py: subprocess

ehavazli committed:
Codacy review:

removed italic style from "other community members" at the end of the file

sssangha committed:
Script crashed when attempting to load generated script. Must add to sys path. Also delete temp file/pycache

sssangha commented:
@ehavazli @dbekaert ok, I tested the updates and fixed a bug in ariaDownload.

Everything works fine now on my end!

dbekaert commented:
ok lets go ahead an merge in dev.
Thank you both for the iterations.
